### PR TITLE
Update for CakePHP 4.x.

### PR DESCRIPTION
**Pull request by dereuromark**
I would recommend we first make a "cake3" branch of the current master.
Then, once all is green here, we can merge to master and make this the new CakePHP 4 compatible branch and also tag a new major here.

### CODE REVIEW

**Comment by dereuromark** [comment]
>  Problem 1
    - php-http/client-integration-tests v0.6.2 requires phpunit/phpunit ^4.8.35 || ^5.4.3 || ^6.0 ->

Why do those require phpunit, shouldnt that be require dev?
We need ^8.5 here for this package now ideally.

**Comment by sagikazarmark** [comment]
There is a phpunit8 branch in that package, but it's not merged yet.

**Comment by dereuromark** [comment]
OK, we will have to wait for that to be merged then before we can continue here.

**Comment by sagikazarmark** [comment]
Feel free to use it for now.

**Comment by dereuromark** [comment]
Can we tag a 0.3.0-beta ?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Change Log
 
+## 0.3.0
```
Comment (sagikazarmark) [inline CHANGELOG.md:3]:
```suggestion
## Unreleased
```

**Comment by sagikazarmark** [comment]
I guess we should merge this PR first and then tag a beta version.

**Comment by dereuromark** [comment]
What do we do with
```
Problem 1
    - php-http/client-integration-tests v0.6.2 requires phpunit/phpunit ^4.8.35 || ^5.4.3 || ^6.0 -> satisfiable by phpunit/phpunit[4.8.35, 4.8.36, 5.4.3, 5.4.4, 5.4.5, 5.4.6, 5.4.7, 5.4.8, 5.5.0, 5.5.1, 5.5.2, 5.5.3, 5.5.4, 5.5.5, 5.5.6, 5.5.7, 5.6.0, 5.6.1, 5.6.2, 5.6.3, 5.6.4, 5.6.5, 5.6.6, 5.6.7, 5.6.8, 5.7.0, 5.7.1, 5.7.10, 5.7.11, 5.7.12, 5.7.13, 5.7.14, 5.7.15, 5.7.16, 5.7.17, 5.7.18, 5.7.19, 5.7.2, 5.7.20, 5.7.21, 5.7.22, 5.7.23, 5.7.24, 5.7.25, 5.7.26, 5.7.27, 5.7.3, 5.7.4, 5.7.5, 5.7.6, 5.7.7, 5.7.8, 5.7.9, 6.0.0, 6.0.1, 6.0.10, 6.0.11, 6.0.12, 6.0.13, 6.0.2, 6.0.3, 6.0.4, 6.0.5, 6.0.6, 6.0.7, 6.0.8, 6.0.9, 6.1.0, 6.1.1, 6.1.2, 6.1.3, 6.1.4, 6.2.0, 6.2.1, 6.2.2, 6.2.3, 6.2.4, 6.3.0, 6.3.1, 6.4.0, 6.4.1, 6.4.2, 6.4.3, 6.4.4, 6.5.0, 6.5.1, 6.5.10, 6.5.11, 6.5.12, 6.5.13, 6.5.14, 6.5.2, 6.5.3, 6.5.4, 6.5.5, 6.5.6, 6.5.7, 6.5.8, 6.5.9] but these conflict with your requirements or minimum-stability.
    - php-http/client-integration-tests v0.6.1 requires phpunit/phpunit ^4.8.35 || ^5.4.3 || ^6.0 -> satisfiable by phpunit/phpunit[4.8.35, 4.8.36, 5.4.3, 5.4.4, 5.4.5, 5.4.6, 5.4.7, 5.4.8, 5.5.0, 5.5.1, 5.5.2, 5.5.3, 5.5.4, 5.5.5, 5.5.6, 5.5.7, 5.6.0, 5.6.1, 5.6.2, 5.6.3, 5.6.4, 5.6.5, 5.6.6, 5.6.7, 5.6.8, 5.7.0, 5.7.1, 5.7.10, 5.7.11, 5.7.12, 5.7.13, 5.7.14, 5.7.15, 5.7.16, 5.7.17, 5.7.18, 5.7.19, 5.7.2, 5.7.20, 5.7.21, 5.7.22, 5.7.23, 5.7.24, 5.7.25, 5.7.26, 5.7.27, 5.7.3, 5.7.4, 5.7.5, 5.7.6, 5.7.7, 5.7.8, 5.7.9, 6.0.0, 6.0.1, 6.0.10, 6.0.11, 6.0.12, 6.0.13, 6.0.2, 6.0.3, 6.0.4, 6.0.5, 6.0.6, 6.0.7, 6.0.8, 6.0.9, 6.1.0, 6.1.1, 6.1.2, 6.1.3, 6.1.4, 6.2.0, 6.2.1, 6.2.2, 6.2.3, 6.2.4, 6.3.0, 6.3.1, 6.4.0, 6.4.1, 6.4.2, 6.4.3, 6.4.4, 6.5.0, 6.5.1, 6.5.10, 6.5.11, 6.5.12, 6.5.13, 6.5.14, 6.5.2, 6.5.3, 6.5.4, 6.5.5, 6.5.6, 6.5.7, 6.5.8, 6.5.9] but these conflict with your requirements or minimum-stability.
    - php-http/client-integration-tests v0.6.0 requires phpunit/phpunit ^4.8.35 || ^5.4.3 || ^6.0 -> satisfiable by phpunit/phpunit[4.8.35, 4.8.36, 5.4.3, 5.4.4, 5.4.5, 5.4.6, 5.4.7, 5.4.8, 5.5.0, 5.5.1, 5.5.2, 5.5.3, 5.5.4, 5.5.5, 5.5.6, 5.5.7, 5.6.0, 5.6.1, 5.6.2, 5.6.3, 5.6.4, 5.6.5, 5.6.6, 5.6.7, 5.6.8, 5.7.0, 5.7.1, 5.7.10, 5.7.11, 5.7.12, 5.7.13, 5.7.14, 5.7.15, 5.7.16, 5.7.17, 5.7.18, 5.7.19, 5.7.2, 5.7.20, 5.7.21, 5.7.22, 5.7.23, 5.7.24, 5.7.25, 5.7.26, 5.7.27, 5.7.3, 5.7.4, 5.7.5, 5.7.6, 5.7.7, 5.7.8, 5.7.9, 6.0.0, 6.0.1, 6.0.10, 6.0.11, 6.0.12, 6.0.13, 6.0.2, 6.0.3, 6.0.4, 6.0.5, 6.0.6, 6.0.7, 6.0.8, 6.0.9, 6.1.0, 6.1.1, 6.1.2, 6.1.3, 6.1.4, 6.2.0, 6.2.1, 6.2.2, 6.2.3, 6.2.4, 6.3.0, 6.3.1, 6.4.0, 6.4.1, 6.4.2, 6.4.3, 6.4.4, 6.5.0, 6.5.1, 6.5.10, 6.5.11, 6.5.12, 6.5.13, 6.5.14, 6.5.2, 6.5.3, 6.5.4, 6.5.5, 6.5.6, 6.5.7, 6.5.8, 6.5.9] but these conflict with your requirements or minimum-stability.
    - Installation request for php-http/client-integration-tests ^0.6 -> satisfiable by php-http/client-integration-tests[v0.6.0, v0.6.1, v0.6.2].
```

They block green travis and the merge.

**Comment by dereuromark** [comment]
Code is working fine. 
We can probably try to fix up the failing test(suite) along the way.

Shall we can tag a beta, rc, or even stable release?

**Comment by dereuromark** [comment]
I did a https://github.com/php-http/cakephp-adapter/releases/tag/0.3.0-beta
soon we should then release a 0.3.0 stable.

**Pull request merged**